doc/contributing/release-process.rst

```diff
@@ -7,517 +7,502 @@ new CKAN release.
+with patches and tested (this will probably be longer for bigger releases).
 
-     But dread found changed the first step slightly to get it to work::
+.. note:: The following steps might need to be repeated at various times to ensure the branch is up to date.
 
-        git log --pretty=format:%s --reverse --no-merges release-v2.4.2...release-v2.5.0 -- | grep -Pzo "^\[#\K[0-9]+" | sort -u -n > issues_2.5.txt
+* **Backports** to the release branch are done via the :ref:`automated backports action <automated_backports>`
+  whenever possible. If there are conflicts, the relevant commits need to be
+  manually cherry-picked.
 
-     When all fragments are ready, make a test build::
+* If there are **security patches** that need to be applied there needs to be a pull request
+  targeting the release branch in the private advisory fork (in addition to the one
+  targeting master). **Do not merge those** until just before the release, otherwise we will
+  publicise vulnerabilities, but allow some time to fix potential issues after merging.
+  Request CVE identifiers with enough time so they are ready on release day (they might
+  take a couple of days to be allocated)
 
-        towncrier build --draft
-
-     And check output. If no problems identified, compile updated
-     changelog::
-
-        towncrier build --yes
+* Check if there are **requirements** that need to be upgraded because of security issues. Check the
+  relevant branch on Snyk_ to see the vulnerable packages. We only upgrade those that don't introduce
+  backwards incompatible changes. In general, upgrading a Python package is just a matter of
+  bumping the version number in ``requirements.in`` and running::
 
-     You'll be asked, whether it's ok to remove source fragments. Feel
-     free to answer "yes" - all changes will be automatically inserted
-     into changelog, so there is no sense in keeping those
-     files. Don't forget to commit changes afterwards.
+      pip-compile -P <package_name> requirements.in
 
-#. A week before the translations will be closed send a reminder email.
-
-#. Once the translations are closed, sync them from Transifex.
+   Make sure to also update ``package.json`` for security related upgrades.
 
-   Pull the updated strings from Transifex::
-
-        tx pull --all --minimum-perc=5 --force
-
-   Check and compile them as before::
+* Pull the latest **translations** from Transfiex and compile them (it's best to split it
+  in two separate commits)::
 
-        ckan -c |ckan.ini| translation check-po ckan/i18n/*/LC_MESSAGES/ckan.po
-        python setup.py compile_catalog
-
-    The compilation shows the translation percentage. Compare this with the new
-    languages directories added to ckan/i18n::
-
-        git status
-
-   ``git add`` any new ones. (If all is well, you won't see any that are under
-   5% translated.)
+      tx pull --all --minimum-perc=5 --force
```

Comment:
include examples of `git commit` with typical commit messages here since it's mentioned above.